api/api.go

```diff
@@ -1090,16 +1099,28 @@ func encodeBody(obj interface{}) (io.Reader, error) {
 
 // requireOK is used to wrap doRequest and check for a 200
 func requireOK(d time.Duration, resp *http.Response, e error) (time.Duration, *http.Response, error) {
+	return requireHttpCodes(d, resp, e, []int{200})
+}
+
+// requireHttpCodes checks for the "allowable" http codes for a response
+func requireHttpCodes(d time.Duration, resp *http.Response, e error, httpCodes []int) (time.Duration, *http.Response, error) {
```

Comment:
very minor: Using a varargs for `httpCodes` might be nice:
```suggestion
func requireHttpCodes(d time.Duration, resp *http.Response, e error, httpCodes ...int) (time.Duration, *http.Response, error) {
```

That would allow the caller to do things like:

```go
requireHttpCodes(d, resp, e, 200)
requireHttpCodes(d, resp, err, 200, 429)
```

but maybe that's just personal preference, up to you

Reply:
That's fair -- I added this!